salt/modules/cmdmod.py

```diff
@@ -414,8 +414,10 @@ def _get_stripped(cmd):
         # method doesn't work properly when run as root for certain commands.
         if isinstance(cmd, (list, tuple)):
             cmd = ' '.join(map(_cmd_quote, cmd))
+            # This patch is explained on https://github.com/saltstack/salt/pull/52632#issuecomment-485046644
+            cmd = cmd.replace("'", '"')
 
-        cmd = 'su -l {0} -c "cd {1}; {2}"'.format(runas, cwd, cmd)
+        cmd = "su -l {0} -c 'cd {1} && {2}'".format(runas, cwd, cmd)
```

Comment:
A better fix would not to have this run by a shell.  I suspect we have 
`sh -c "su -l {0} -c 'cd {1} && {2}'"` happening.
The code should assume the su will invoke the shell and skip prefix with `sh -c` some where later in the code.

Reply:
This is a good point, unfortunately I don't have too much time to investigate this in a short time.

If you prefer not to approve this merge I will understand it, although this change solves a security bug.

Reply:
Finally I have used `_cmd_quote` to prevent shell injection. (`_cmd_quote` is an alias for `shlex.quote` which handles single quotes properly for shell commands)